File: vw_rust/src/window/sdl_initializer.rs
use crate::sys::bindings;

pub struct SdlInitializer {
    pub ptr: *mut bindings::vw_SDL_Initializer,
}

impl SdlInitializer {
    pub fn new() -> SdlInitializer {
        unsafe {
            SdlInitializer {
                ptr: bindings::vw_create_SDL_Initializer(),
            }
        }
    }
}

impl Drop for SdlInitializer {
    fn drop(&mut self) {
        unsafe {
            bindings::vw_destroy_SDL_Initializer(self.ptr);
        }
    }
}

File: vw_c/src/Window/SDL_Initializer.cpp
#include "Window/SDL_Initializer.h"

#include <VulkanWrapper/Window/SDL_Initializer.h>

vw::SDL_Initializer *vw_create_sdl_initializer() {
    return new vw::SDL_Initializer;
}

void vw_destroy_sdl_initializer(vw::SDL_Initializer *ptr) { delete ptr; }
